platform/viewer/src/connectedComponents/Viewer.js

```diff
@@ -365,18 +368,36 @@ const _mapStudiesToThumbnails = function(studies) {
   return studies.map(study => {
     const { StudyInstanceUID } = study;
 
+    let finishedProcessing = true;
+
     const thumbnails = study.displaySets.map(displaySet => {
       const {
         displaySetInstanceUID,
         SeriesDescription,
         SeriesNumber,
         InstanceNumber,
         numImageFrames,
+        SeriesDate,
+        SeriesTime,
       } = displaySet;
 
       let imageId;
       let altImageText;
 
+      let seriesDateTime = '';
+
+      if (SeriesDate) {
+        if (SeriesTime) {
+          seriesDateTime = `${SeriesDate}${SeriesTime}`;
+        } else {
+          seriesDateTime = `${SeriesDate}$`;
+        }
+      }
+
+      if (!displaySet.hasOwnProperty('SeriesDate')) {
```

Comment:
Happy for a better suggestion about how to do this @dannyrb .

The sorting is expensive and this is called many times on load whilst each displaySet is processed. The placeholder objects don't have a SeriesDate, and all displaySets do once processed. DisplaySet creation can be asynchronus so its a little sticky.

Reply:
This is now outdated.